layer_test.go

```diff
@@ -881,4 +792,64 @@ func testLayer(t *testing.T, context spec.G, it spec.S) {
 			Expect(string(data)).To(ContainSubstring(`"Source":{`))
 		})
 	})
+
+	context("ContributableBuildFunc", func() {
+		it("calls contributeLayersFunc and processes each layer contributor", func() {
+			mockContributor1 := &mocks.Contributable{}
+			mockContributor2 := &mocks.Contributable{}
+
+			mockContributor1.On("Name").Return("layer-1")
+			mockContributor2.On("Name").Return("layer-2")
+
+			mockContributor1.On("Contribute", mock.Anything).Return(nil)
+			mockContributor2.On("Contribute", mock.Anything).Return(nil)
+
+			buildFunc := libpak.ContributableBuildFunc(func(context libcnb.BuildContext, result *libcnb.BuildResult) ([]libpak.Contributable, error) {
+				return []libpak.Contributable{
+					mockContributor1,
+					mockContributor2,
+				}, nil
+			})
+
+			buildResult, err := buildFunc(libcnb.BuildContext{})
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(buildResult.Layers[0].Name).To(Equal("layer-1"))
+			Expect(buildResult.Layers[1].Name).To(Equal("layer-2"))
+			Expect(mockContributor1.Calls).To(HaveLen(2))
```

Comment:
What is this expected to verify? Probably that `Contribute` is called, right?

Reply:
This is a little painful to test. The ContributableBuildFunc takes in the mock Contributables which in turn get cycled through. This is trying to verify that we ran them and that it used the name in the expected way, which is to set the name on the layer produced by ContributableBuildFunc. The HaveLen check is to make sure that it called both Name and Contribute.

Probably other ways to test this. If there's a preference for doing it a different way, in particular one that's more clear, I'm fine with that. I have no particular attachment to this approach.